### ISSUE TITLE
About the result of "EEGClip"

### ISSUE THREAD

**Issue by xuxiran** [prajwalsingh/EEGStyleGAN-ADA#9]
Thank you for your excellent work and patient answers before.  I tried many times but I failed, so I came again. In the "EEGClip", I have two questions.
The first is about the result. I have run the "main.py" for 2785 epochs but the result is still lower than 0.05.
![image](https://github.com/prajwalsingh/EEGStyleGAN-ADA/assets/48015859/655bd9f8-9f7d-4497-add6-a8ee993e6155)
The second is about evaluating. I could only find "main.py" to train the model but I did not find "evaluate.py" to get the result of testing data. Perhaps I overlooked something?
Thank you very much for your attention and patience!

**Comment by prajwalsingh** [comment]
Hi @xuxiran , check this issue [ [link](https://github.com/prajwalsingh/EEGStyleGAN-ADA/issues/2#issuecomment-1891597052) ].  If it doesn't work for you, please let us know.

For evaluation this is code we have used [ [topkrecall.py](https://github.com/prajwalsingh/EEGStyleGAN-ADA/blob/main/EEGClip/topkrecall.py) ]

**Comment by xuxiran** [comment]
Thank you very much for your timely response. I have read the closed issue #2 and #7. I made sure that the dataset is "5-95Hz". However, I could not get satisfactory answer.
I doubt that maybe I miss some pre-programs? Do I need to prepare any other files before running "main. py" in "EEGClip"? Just like before running "train. py" for "EEGStyleGAN-ADA-CVPR40", I need to prepare eegbestckpt and imageckpt. 
Or, as I understand it, "EEGClip" should be an end-to-end architecture, and I should simply run "main. py"?


In addition, I evaluated this code by running "topkrecall.py", but I still could not get a good result. After modifying some problems, I successfully ran the "topkrecall.py", but the result was still lower than I expected.
![image](https://github.com/prajwalsingh/EEGStyleGAN-ADA/assets/48015859/aa8bb5cc-29a1-4cce-aaf0-7d21a6aed02a)
When I try to use the pre-trained parameters, it seems that the model of pre-trained parameters is not the model we got in "EEGClip".
![image](https://github.com/prajwalsingh/EEGStyleGAN-ADA/assets/48015859/16e97a2b-2549-4fd4-9925-fcbc3511d381)
Thank you very much for your attention and patience!

**Comment by prajwalsingh** [comment]
Hi @xuxiran , thank you for raising the issue. I am adding Dwip in the loop @dwipddalal in the loop. He will address the problem soon.

**Comment by xuxiran** [comment]
Thank you very much! You are the best repository owner I have seen!

**Comment by xuxiran** [comment]
![image](https://github.com/prajwalsingh/EEGStyleGAN-ADA/assets/48015859/5404b2f3-9e08-41d5-8115-ccca705631ff)
I have finished running main.py and are running classification.py now. It seems works. Thank you very much!

**Comment by zihang-xu** [comment]
Congratulations! You seem to be almost successful.
Do you use filtered EEG data? I'm still running main.py. Can you share the clip_2048.pth file?

**Comment by prajwalsingh** [comment]
Hi @zihang-xu , we are also running the experiments in parallel. We will upload the clip_2048.pth once it is done. We made a few updates to the code after making the final updates on Git Hub. I will let you know. 

For filtered EEG data, we have to check again. We have used and reported the results in the paper.

P.S. The initial experiment code got deleted due to some confusion while cleaning the drive, so we have to re-run and check which one worked earlier 😅

**Comment by zihang-xu** [comment]
Thanks for your patience and explanation! I look forward to your updates. 

Experiment results can be deleted by mistake. I've done the same thing 🤣

**Comment by zihang-xu** [comment]
If you have new results, look forward to sharing and updating them. :)
Thanks again！

**Comment by weipipione** [comment]
Is EEGCLIP also a zero sample recognition task?

**Comment by prajwalsingh** [comment]
Hi @weipipione , if you are asking about zero-shot classification. We have performed that experiment with a triplet-loss-based feature extraction method. We haven't explored it for EEGCLIP.

**Comment by Applicant-233** [comment]
@prajwalsingh Hello, may I ask where the original data was obtained? I did not find any relevant connections, like EEGCVPR40 Dataset

**Comment by prajwalsingh** [comment]
Hi @Applicant-233 , the data is obtained from here [[link](https://github.com/perceivelab/eeg_visual_classification?tab=readme-ov-file)]

**Comment by Applicant-233** [comment]
> Hi @Applicant-233 , the data is obtained from here [[link](https://github.com/perceivelab/eeg_visual_classification?tab=readme-ov-file)]

Thank you! The link offers raw data, but the preprocessed dataset used in the paper is not included.

**Comment by prajwalsingh** [comment]
@Applicant-233 Both raw and preprocess data are there with name 5_95...

**Comment by Applicant-233** [comment]
> @Applicant-233 Both raw and preprocess data are there with name 5_95...

 Thanks!

**Comment by LuigiSigillo** [comment]
Thank you for your work, so the checkpoint for the model present in EEGStyleGAN-ADA/EEGClip/classification_eeg.py is not uploaded yet? I cant find it on the repo

**Comment by aohodo** [comment]
这个代码从哪里运行呢
